packages/pressure/src/monitor.ts

```diff
@@ -0,0 +1,81 @@
+import { defaults } from '@directus/utils';
+import type { IntervalHistogram } from 'node:perf_hooks';
+import { monitorEventLoopDelay, performance } from 'node:perf_hooks';
+import { memoryUsage } from 'node:process';
+import { setTimeout } from 'node:timers';
+
+export type PressureMonitorOptions = {
+	maxEventLoopDelay?: number | false;
+	maxEventLoopUtilization?: number | false;
+	maxMemoryHeapUsed?: number | false;
+	maxMemoryRss?: number | false;
+	sampleInterval?: number;
+	resolution?: number;
+};
+
+export class PressureMonitor {
+	private memoryHeapUsed = 0;
+	private memoryRss = 0;
+	private eventLoopDelay = 0;
+	private eventLoopUtilization = 0;
+	private options: Required<PressureMonitorOptions>;
+	private histogram: IntervalHistogram;
+	private timeout: NodeJS.Timeout;
+
+	constructor(options: PressureMonitorOptions = {}) {
+		this.options = defaults(options, {
+			sampleInterval: 250,
+			resolution: 10,
+			maxMemoryHeapUsed: false,
+			maxMemoryRss: false,
+			maxEventLoopDelay: false,
+			maxEventLoopUtilization: false,
+		});
+
+		this.histogram = monitorEventLoopDelay({ resolution: this.options.resolution });
+		this.histogram.enable();
+
+		this.updateUsage = this.updateUsage.bind(this);
+		this.timeout = setTimeout(this.updateUsage, this.options.sampleInterval);
```

Comment:
```suggestion
		this.timeout = setTimeout(this.updateUsage, Number(this.options.sampleInterval));
```
I think we should also do some more sanity checking beyond just casting it as a number because `setTimeout` accepts strings and infinity's without error but silently falls back to zero or one as timeout 😬 meaning the `updateUsage` function becomes an infinite loop firing every millisecond.